src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py

```diff
@@ -89,41 +92,63 @@ def load_arguments(self, _):
 
     with self.argument_context('policy assignment create', resource_type=ResourceType.MGMT_RESOURCE_POLICY) as c:
-        c.argument('name', options_list=('--name', '-n'), help='name of the new assignment')
-        c.argument('params', options_list=('--params', '-p'), help='JSON formatted string or path to file with parameter values of policy rule', min_api='2016-12-01')
+        c.argument('name', options_list=('--name', '-n'), help='Name of the new policy assignment.')
+        c.argument('params', options_list=('--params', '-p'), help='JSON formatted string or path to file with parameter values of policy rule.', min_api='2016-12-01')
 
     with self.argument_context('policy assignment create', resource_type=ResourceType.MGMT_RESOURCE_POLICY, min_api='2017-06-01-preview') as c:
-        c.argument('policy_set_definition', options_list=('--policy-set-definition', '-d'), help='name or id of the policy set definition.')
+        c.argument('policy_set_definition', options_list=('--policy-set-definition', '-d'), help='Name or id of the policy set definition.')
         c.argument('sku', options_list=('--sku', '-s'), help='policy sku.', arg_type=get_enum_type(['free', 'standard']))
         c.argument('notscopes', options_list=('--not-scopes'), nargs='+')
 
     with self.argument_context('policy set-definition', min_api='2017-06-01-preview', resource_type=ResourceType.MGMT_RESOURCE_POLICY) as c:
+        c.argument('display_name', help='Display name of policy set definition.')
+        c.argument('description', help='Description of policy set definition.')
+        c.argument('params', help='JSON formatted string or a path to a file or uri with parameter definitions.', type=file_type, completer=FilesCompleter())
+        c.argument('definitions', help='JSON formatted string or a path to a file or uri containing definitions.', type=file_type, completer=FilesCompleter())
+        c.argument('subscription_id', arg_type=subscription_id_type)
+        c.argument('management_group', arg_type=management_group_name_type)
+
+    with self.argument_context('policy set-definition create', min_api='2017-06-01-preview', resource_type=ResourceType.MGMT_RESOURCE_POLICY) as c:
+        c.argument('name', options_list=('--name', '-n'), help='Name of the new policy set definition.')
+
+    with self.argument_context('policy set-definition show', min_api='2017-06-01-preview', resource_type=ResourceType.MGMT_RESOURCE_POLICY) as c:
+        c.argument('policy_set_definition_name', arg_type=existing_policy_set_definition_name_type)
+
+    with self.argument_context('policy set-definition update', min_api='2017-06-01-preview', resource_type=ResourceType.MGMT_RESOURCE_POLICY) as c:
+        c.argument('policy_set_definition_name', arg_type=existing_policy_set_definition_name_type)
+
+    with self.argument_context('policy set-definition delete', min_api='2017-06-01-preview', resource_type=ResourceType.MGMT_RESOURCE_POLICY) as c:
```

Comment:
Same here. This code duplication should be eliminated.

Reply:
Done.